cmd/prototype.go

```diff
@@ -397,7 +398,11 @@ func expandPrototype(proto *prototype.SpecificationSchema, templateType prototyp
 		return "", err
 	}
 	if templateType == prototype.Jsonnet {
-		template = append([]string{`local params = std.extVar("` + metadata.ParamsExtCodeKey + `").components.` + componentName + ";"}, template...)
+		componentsText := "components." + componentName
+		if utils.HasSpecialCharacter(componentName) {
+			componentsText = fmt.Sprintf(`components["%s"]`, componentName)
+		}
+		template = append([]string{`local params = std.extVar("` + metadata.ParamsExtCodeKey + `").` + componentsText + ";"}, template...)
```

Comment:
We could just simplify this by using `components["whatever"]` syntax for _every_ component, no? The way we're doing it now is cleaner, so i think it's nbd either way, just thought I'd bring up the alternatives and see what you think.

Reply:
Think I'm going to leave this since `components.whatever` is cleaner. Prefer to use that syntax where we can.